src/api/composition/DefaultCompositionProvider.js

```diff
@@ -68,45 +65,78 @@ define([
     /**
      * Load any domain objects contained in the composition of this domain
      * object.
-     * @param {module:openmct.DomainObjcet} domainObject the domain object
+     * @param {module:openmct.DomainObject} domainObject the domain object
      *        for which to load composition
-     * @returns {Promise.<Array.<module:openmct.DomainObject>>} a promise for
-     *          the domain objects in this composition
+     * @returns {Promise.<Array.<module:openmct.Identifier>>} a promise for
+     *          the Identifiers in this composition
      * @memberof module:openmct.CompositionProvider#
      * @method load
      */
     DefaultCompositionProvider.prototype.load = function (domainObject) {
-        return Promise.all(domainObject.composition.map(ObjectAPI.get));
+        return Promise.all(domainObject.composition);
     };
 
+    /**
+     * Attach listeners for changes to the composition of a given domain object.
+     * Supports `add` and `remove` events.
+     *
+     * @param {module:openmct.DomainObject} domainObject to listen to
+     * @param String event the event to bind to, either `add` or `remove`.
+     * @param Function callback callback to invoke when event is triggered.
+     * @param [context] context to use when invoking callback.
+     */
     DefaultCompositionProvider.prototype.on = function (
         domainObject,
         event,
-        listener,
+        callback,
         context
```

Comment:
Out of context here, but this is a particularly useful style.

Binding functions means you have to save a reference to the bound function so you can remove the listener at a later date.  This simplifies that process; it also mimics the eventemitter interface, which has the same behavior.

In a similar vein, all the iterator functions (`Array.prototype.map`, `filter`, `forEach`, etc) accept a context argument.  It's also one of my biggest gripes with lodash, because they thought being a pure functional library was more important than being a useful library, and they don't support the context argument.